src/Aspire.Hosting/Dashboard/ContainerSnapshot.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using Aspire.Dashboard.Model;
+using Google.Protobuf.WellKnownTypes;
+
+namespace Aspire.Hosting.Dashboard;
+
+/// <summary>
+/// Immutable snapshot of a container's state at a point in time.
+/// </summary>
+public sealed class ContainerSnapshot : ResourceSnapshot
+{
+    public override string ResourceType => KnownResourceTypes.Container;
+
+    public required string? ContainerId { get; init; }
+    public required string Image { get; init; }
+    public required ImmutableArray<int> Ports { get; init; }
```

Comment:
Is there any reason this needs to be immutable? Isn't this just readonly? This snapshot is never updated right?

Reply:
I like immutable arrays for their compactness in memory and efficiency of operations on them. Do you have any concerns?

Reply:
I prefer using interfaces over immutable collections if we're not taking advantage of the immutability.

Reply:
What about making these types internal? Then the collection type is something we're free to change. I don't think these types will be needed outside of the app host, in the end.

Reply:
> I prefer using interfaces over immutable collections

@davidfowl what interface would you prefer here? `ICollection`, `IReadOnlyList`, something else? I'm curious.

Reply:
I'm assuming it's `IReadOnlyList<T>` which would make sense for public properties, as it gives us the flexibility to swap out the implementation at some later point. I propose making this internal (which is possible given the split between projects) and keeping `ImmutableArray<>` because it:

- Takes less memory
- Is less indirect than `List<>` as there are fewer pointer follows
- Doesn't allocate when you enumerate
- Has built-in extension methods that implement Linq-isms more cheaply than when using interfaces

Reply:
Make it internal and use the best type (probably not an interface).

In the future if it needs to be public then we can review the API in detail and make changes.

Reply:
I've made all these types internal.